packages/libs/src/sdk/services/Solana/programs/CustomInstructionError.ts

```diff
@@ -0,0 +1,35 @@
+import { SendTransactionError } from '@solana/web3.js'
+
+type CustomInstructionErrorMessage = {
+  InstructionError: [number, { Custom: number }]
+}
+
+export class CustomInstructionError extends Error {
+  constructor(public instructionIndex: number, public code: number) {
+    super(
+      JSON.stringify({
+        InstructionError: [instructionIndex, { Custom: code }]
+      })
+    )
+  }
+
+  /**
+   * @throws if the error message isn't a custom error
```

Comment:
does this throw?

Reply:
It used to 😅 I went back and forth on having it throw or not. Outdated comment - good catch!